opal/corelib/range.rb

```diff
@@ -126,18 +146,55 @@ def size
   end
 
   def step(n = 1)
-    raise NotImplementedError
+    return enum_for(:step, n) unless block_given?
+    raise ArgumentError.new("step can't be negative") if n < 0
+    raise ArgumentError.new("step can't be 0") unless n > 0
+    if @begin.is_a?(Numeric) and @end.is_a?(Numeric)
```

Comment:
For what it's worth MRI uses `rb_obj_is_kind_of ` and rubinius uses `kind_of?` checks here (`is_a?` and `kind_of?` are synonyms). Both of them can be overriden, this seems closer to semantics of other rubies, even if specs don't check it.